crypto/storage/fs.go

```diff
@@ -75,12 +77,26 @@ func NewFileSystemBackend(fspath string) (Storage, error) {
 	return fsc, nil
 }
 
+func (fsc *fileSystemBackend) ListPrivateKeys() []string {
+	var result []string
+	_ = filepath.Walk(fsc.fspath, func(path string, info fs.FileInfo, err error) error {
+		if !info.IsDir() && strings.HasSuffix(info.Name(), string(privateKeyEntry)) {
+			upper := len(info.Name()) - len(privateKeyEntry) - 1
+			if upper < len(info.Name()) {
```

Comment:
If we do this:

`upper := len(info.Name()) - len(privateKeyEntry) - 1`

Shouldn't `upper` always be less than `len(info.Name())` (because of the `- 1` part)?